src/som/vmobjects/primitive.py

```diff
@@ -72,6 +72,10 @@ def invoke_n(self, stack, stack_ptr):
         prim_fn = self._prim_fn
         return prim_fn(self, stack, stack_ptr)
 
+    def invoke_args(self, rcvr, args):
```

Comment:
Hm, and this works?

Isn't the interface to the BcPrimitives different? looks like it has a `invoke_n` method?

Reply:
This was to avoid creating a new instance of `CachedDispatchNode`, since we rely on it in the BC interp now and it does a call to `invoke_args` and not `invoke_n`. This was a bandaid fix, the best solution would be to make it so that both BC and AST use the same naming scheme for this operation to avoid needless duplication. `invoke_n` sounds better to me personally.

Reply:
The arguments to the functions are not the same.
I think it's easier to reason about if they then do not have the same name.
In the bytecode interpreter, the invoke methods do directly access the stack to avoid creating a new args array, if I remember correctly.

That works at least for the `_1`, `_2`, ... primitives, I guess.
So, I was wondering where where the args array is created for the call, maybe I just missed that place.

Reply:
Fair enough. And no args array is created, it's once again accessing the stack as you described. `invoke_args` seems to only get invoked for `Bytecodes.q_super_send_n`, so I can just remove both that call and that method in `BcPrimitive` entirely by calling `dispatch_node.get_cached_method().invoke_n(stack, stack_ptr)` instead of `dispatch_node.invoke_args(stack, stack_ptr)`, anyway.